convert2rhel/unit_tests/cert_test.py

```diff
@@ -70,20 +78,53 @@ def test_get_cert_path_missing_cert(self):
     @unit_tests.mock(utils, "DATA_DIR", unit_tests.NONEXISTING_DIR)
     @unit_tests.mock(system_info, "arch", "nonexisting_arch")
     def test_get_cert_path_nonexisting_dir(self):
-        self.assertRaises(SystemExit, cert.SystemCert._get_cert_path)
+        self.assertRaises(SystemExit, cert.SystemCert._get_cert)
         self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
 
-    @unit_tests.mock(cert.SystemCert, "_system_cert_dir", unit_tests.TMP_DIR)
     @unit_tests.mock(tool_opts, "arch", "x86_64")
     @unit_tests.mock(utils, "DATA_DIR", os.path.join(base_data_dir, "6", "x86_64"))
     def test_install_cert(self):
         # By initializing the cert object we get a path to an existing
         # certificate based on the mocked parameters above
         system_cert = cert.SystemCert()
+        system_cert._target_cert_dir = unit_tests.TMP_DIR
 
         system_cert.install()
 
-        cert_filename = os.path.basename(system_cert._cert_path)
-        installed_cert_path = os.path.join(system_cert._system_cert_dir, cert_filename)
+        installed_cert_path = os.path.join(system_cert._target_cert_dir, system_cert._cert_filename)
         self.assertTrue(os.path.exists(installed_cert_path))
         shutil.rmtree(unit_tests.TMP_DIR)
+
+
+@pytest.mark.parametrize(
+    ("directory", "filename"),
+    ((unit_tests.TMP_DIR, "existing.file"), (unit_tests.TMP_DIR, unit_tests.NONEXISTING_FILE), (None, None)),
+)
+def test_remove(monkeypatch, directory, filename):
```

Comment:
nit: pytest has fixtures `tmpdir` and `caplog`, try to adopt it, the solution should be more readable and simplier

Reply:
@ZhukovGreen  I've got a problem with 'tmpdir' on older version of python (while running 'make tests' the first one is python2.7.5). I think it's due this code you've shared with me [conftest.py#L22](https://github.com/oamg/convert2rhel/blob/5630fa7b070b0527e0a6426148ce6823108f7610/convert2rhel/unit_tests/conftest.py#L22) because it returns string on older versions, but str hasn't got methods for [creating](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture) file :smiley: 

The test is working correctly while running it on newer compatible version with 'pytest -k <name_of_test>'.

I think, that I could use the 'tmpdir' as the 'unit_tests.TMP_DIR' which I'm using now and it should solve my problem. But with this will go away the advantage of 'tmpdir' fixture.

Reply:
This fixture was created to solve a problem of py26 compatibility. It is no longer the care and we are not running unit tests on this system anymore. Try to remove this fixture

Reply:
@ZhukovGreen even worse. Too many type errors `TypeError: coercing to Unicode: need string or buffer, LocalPath found`

Reply:
Try to troubleshoot this. If dead end, I'll help on Thu-Fri